### PR TITLE
wallet_api: add scanTransactions function

### PR DESCRIPTION
**Pull request by selsta**
Introduced in #7312

Tested on the GUI side and working as expected.

### CODE REVIEW

**Review by tobtoht — APPROVED** [review]

File: src/wallet/api/wallet.cpp [hunk 1]
```diff
@@ -1280,6 +1280,42 @@ bool WalletImpl::importOutputs(const string &filename)
...
+    {
+        crypto::hash txid;
+        if (!epee::string_tools::hex_to_pod(s, txid))
+        {
+            setStatusError(string(tr("Invalid txid specified: ")) + s);
+            return false;
+        }
+        txids_u.insert(txid);
+    }
+    std::vector<crypto::hash> txids_v(txids_u.begin(), txids_u.end());
```
Comment (jeffro256) [inline src/wallet/api/wallet.cpp:1303]: You can skip creating two containers (less memory, less copying) for the hashes if you use a `vector` for `txids_u` and call `.reserve()` before the `for` loop.

Comment (selsta) [reply src/wallet/api/wallet.cpp:1303]: I just copied the simplewallet code. I think unordered set is used to avoid duplicates.

Comment (jeffro256) [reply src/wallet/api/wallet.cpp:1303]: Ah, I see the `dedup` part now, is it bad if a TX gets scanned twice?

Comment (selsta) [reply src/wallet/api/wallet.cpp:1303]: Scanning twice is definitely significantly slower than copying from a set to a vector. I think the current code is fine.

Comment (mj-xmr) [reply src/wallet/api/wallet.cpp:1303]: That's the simplest way to dedup.

File: src/wallet/api/wallet.cpp [hunk 2]
```diff
@@ -1280,6 +1280,42 @@ bool WalletImpl::importOutputs(const string &filename)
...
+    }
+
+    // Parse and dedup args
+    std::unordered_set<crypto::hash> txids_u;
+    for (const auto &s : txids)
+    {
+        crypto::hash txid;
+        if (!epee::string_tools::hex_to_pod(s, txid))
+        {
+            setStatusError(string(tr("Invalid txid specified: ")) + s);
```
Comment (mj-xmr) [inline src/wallet/api/wallet.cpp:1298]: Ignore if you wish but:

`std::string(tr` ?

File: src/wallet/api/wallet2_api.h [hunk 3]
```diff
@@ -927,6 +927,13 @@ struct Wallet
      */
     virtual bool importOutputs(const std::string &filename) = 0;
 
+    /*!
+     * \brief scanTransactions - scan a list of transaction ids, this operation may reveal the txids to the remote node and affect your privacy
+     * \param txids            - list of transaction ids
+     * \return                 - true on success
```
Comment (mj-xmr) [inline src/wallet/api/wallet2_api.h:933]: "True on success, meaning, that all transactions were valid"?

**Review by jeffro256 — APPROVED** [review]

File: src/wallet/api/wallet.cpp (hunk 2 above)
Comment (selsta) [reply src/wallet/api/wallet.cpp:1298]: if I would write new code yes, but every other occurrence in this file uses just `string` so I kept it consistent

File: src/wallet/api/wallet2_api.h (hunk 3 above)
Comment (selsta) [reply src/wallet/api/wallet2_api.h:933]: Success meaning all the transactions got successfully scanned.

**Pull request merged**